common/core/desktop/tests/unit/kmnkbd/test_kmx_xstring.cpp

```diff
@@ -17,6 +17,19 @@
 using namespace km::kbp::kmx;
 using namespace std;
 
+#define U_UC_SENTINEL u"\uFFFF"
+#define U_UC_SURROGATE u"\U0001F609"
+
+#define U_CODE_ANY u"\u0001"
+#define U_CODE_DEADKEY u"\u0008"
+#define U_CODE_IFOPT u"\u0014"
+#define U_CODE_IFSYSTEMSTORE u"\u0017"
+
+#define C_CODE_ANY(store) U_UC_SENTINEL U_CODE_ANY store
+#define C_CODE_IFOPT(opt, val1, val2) U_UC_SENTINEL U_CODE_IFOPT opt val1 val2
+#define C_CODE_IFSYSTEMSTORE(store, val1, val2) U_UC_SENTINEL U_CODE_IFSYSTEMSTORE store val1 val2
+#define C_CODE_DEADKEY(deadkey) U_UC_SENTINEL U_CODE_DEADKEY deadkey
```

Comment:
These look good. I think we should move them to a separate header file, perhaps even into kmx_file.h, below the corresponding integer #defines?

Reply:
I'll do that in the other PR.